packages/postgres/src/postgres-connection-string.ts

```diff
@@ -37,3 +37,29 @@ export function isPooledConnectionString(connectionString: string): boolean {
 export function isDirectConnectionString(connectionString: string): boolean {
   return !isPooledConnectionString(connectionString);
 }
+
+export function isLocalhostConnectionString(connectionString: string): boolean {
+  try {
+    // This seems silly, but we can use all of the hard work put into URL parsing
+    // if we just convert `postgresql://` to `https://` and then parse it as a URL.
+    const withHttpsProtocol = connectionString.startsWith('postgresql://')
```

Comment:
I think this should work without replacing the protocol before parsing the URL:
```
> new URL("postgresql://localhost:5432")
URL {
  href: 'postgresql://localhost:5432',
  origin: 'null',
  protocol: 'postgresql:',
  username: '',
  password: '',
  host: 'localhost:5432',
  hostname: 'localhost',
  port: '5432',
  pathname: '',
  search: '',
  searchParams: URLSearchParams {},
  hash: ''
}
```

Reply:
It does not in some cases 😢 Try running the tests without it.

Reply:
<img width="648" alt="image" src="https://github.com/vercel/storage/assets/76245373/86f27fa0-1e1e-4e7d-9dd4-32c631e441e6">

Actually, this very example doesn't work